nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_ClientService.java

```diff
@@ -212,6 +213,7 @@ public void onEnabled(final ConfigurationContext context) throws InitializationE
             final Admin admin = this.connection.getAdmin();
             if (admin != null) {
                 admin.listTableNames();
+                masterAddress = admin.getClusterStatus().getMaster().getHostAndPort();
```

Comment:
In my fix I used `this.clusterId = admin.getClusterStatus().getClusterId();`, with the assumption that the masterNode could possible between enablements. `clusterId` could potentially remane consistent.

Reply:
@acumartini Thanks for sharing that. I agree with the idea, that ClusterId is more consistent. However, I still prefer using master address as the hostname part of transit URI, because ClusterId is an UUID string, which has less context in itself than master address. But I think having ClusterId in NiFi provenance events would be useful in some situations. I added a note on NIFI-4867 for future improvement. Thanks again!